File: frontend/src/services/commentService.ts
import { api } from "./api";
import { Comment, CommentRequest } from "@/types/comment";

export const CommentService = {
  getByPost: async (postId: number): Promise<Comment[]> => {
    const response = await api.get(`/posts/${postId}/comments`);
    return response.data;
  },
  create: async (postId: number, data: CommentRequest): Promise<Comment> => {
    const response = await api.post(`/posts/${postId}/comments`, data);
    return response.data;
  },
  update: async (id: number, data: CommentRequest): Promise<Comment> => {
    const response = await api.put(`/posts/comments/${id}`, data);
    return response.data;
  },
  delete: async (id: number): Promise<void> => {
    await api.delete(`/posts/comments/${id}`);
  },
};

File: backend/src/main/java/ifsc/edu/tpj/service/CommentService.java
package ifsc.edu.tpj.service;

import ifsc.edu.tpj.dto.CommentRequestDTO;
import ifsc.edu.tpj.model.Comment;
import ifsc.edu.tpj.model.Post;
import ifsc.edu.tpj.repository.CommentRepository;
import ifsc.edu.tpj.repository.PostRepository;
import lombok.RequiredArgsConstructor;

import java.util.List;

import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CommentService {

    private final CommentRepository commentRepository;
    private final UserService userService;
    private final PostRepository postRepository;

    public Comment create(Long postId, CommentRequestDTO dto){

        Post post = postRepository.findById(postId)
                .orElseThrow(()-> new RuntimeException("Post not found: " + postId));
        
        return commentRepository.save(Comment.builder()
                        .author(userService.findUserById(dto.userId()))
                        .body(dto.body()).post(post)
                        .post(post)
                        .build());
    }


    public List<Comment> findByPost(Long postId) {
        return commentRepository.findByPost_PostId(postId);
    }

    public Comment findById(Long id){
        return commentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Comment not found: " + id));
    }

    public Comment update(Long id, CommentRequestDTO dto){
        Comment comment = findById(id);

        if (!comment.getAuthor().getUserId().equals(dto.userId())) {
            throw new RuntimeException("Somente o autor pode editar esse comentário");
        }
        
        comment.setBody(dto.body());
        commentRepository.save(comment);
        return comment;
    }

    public Comment delete(Long id){
        Comment comment = findById(id);
        
        commentRepository.delete(comment);
        return comment;
    }

}
